RxQuiz/ViewController/QuizViewController.swift

```diff
@@ -127,43 +126,57 @@ final class QuizViewController: UIViewController {
         }
         
         currentQuestionIndex.asDriver()
+            .filter { [unowned self] in $0 < self.questions.value.endIndex }
             .driveNext { [unowned self] count in
-                switch count {
-                case self.questions.value.endIndex:
-                    func restart() {
-                        self.questions.value.removeAll()
-                        self.currentQuestionIndex.value = 0
-                        self.correctAnswerCount.value = 0
-                        self.startDate = NSDate()
+                let question = self.questions.value[count]
+                self.currentQuestion.value = question
+                self.currentAnswer.value = question.answer
+            }
+            .addDisposableTo(disposeBag)
+        
+        currentQuestionIndex.asDriver()
+            .filter { [unowned self] in $0 == self.questions.value.endIndex }
+            .withLatestFrom(correctAnswerCount.asDriver()
+                .map { count -> String in
```

Comment:
Please check the following code:
```swift
let OK = [1].map { count in
    count > 5 ? "Good" : "Bad"
}

let NG = [1].map { count in
    let s = count > 5 ? "Good" : "Bad"
    return s
}
// error: expression type '[_]' is ambiguous without more context
```

```swift
let OK = {
    "Test"
}

let NG = {
    let s = "Test"
    return s
}
// error: unable to infer closure return type in current context
```

I think Swift 2.2 compiler requires return type in closure when the closure has multiple lines.